bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/init/NodeInitializers.java

```diff
@@ -0,0 +1,209 @@
+import edu.gemini.spModel.seqcomp.SeqRepeatDarkObs;
+import edu.gemini.spModel.seqcomp.SeqRepeatObserve;
+import edu.gemini.spModel.obslog.ObsExecLog;
+import edu.gemini.spModel.obslog.ObsQaLog;
+import edu.gemini.spModel.target.obsComp.TargetObsComp;
+import edu.gemini.spModel.template.TemplateFolder;
+import edu.gemini.spModel.template.TemplateGroup;
+import edu.gemini.spModel.template.TemplateParameters;
+import edu.gemini.shared.util.immutable.Option;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+
+/**
+ * The collection of all default node initializers used by the science program
+ * factory.
+ */
+public enum NodeInitializers {
+    instance;
+
+    public final ISPNodeInitializer<ISPConflictFolder, ConflictFolder> conflict =
+            ConflictFolderNI.instance;
+
+    public final ISPNodeInitializer<ISPGroup, SPGroup> group =
+            SPGroup.NI;
+
+    public final ISPNodeInitializer<ISPObservation, SPObservation> obsNoInstrument =
+            ObservationNI.NO_INSTRUMENT;
+
+    private final Map<Instrument, ISPNodeInitializer<ISPObservation, SPObservation>> obsInitMap;
+
+    {
+        // Default values, one for each instrument type.
+        final Map<Instrument, ISPNodeInitializer<ISPObservation, SPObservation>> m =
+                Arrays.asList(Instrument.values()).stream().collect(Collectors.toMap(
+                    Function.identity(),
+                    i -> ObservationNI.forInstrument(i)));
+
+        // Replace default values with more specific initializers as necessary.
+        m.put(Instrument.Gsaoi, Gsaoi.OBSERVATION_NI);
+        // m.put(Instrument.Ghost, Ghost.OBSERVATION_NI);
+
+        obsInitMap = Collections.unmodifiableMap(m);
+    }
+
+    public final ISPNodeInitializer<ISPObservation, SPObservation> obs(Option<Instrument> inst) {
+        return inst.map(i -> obsInitMap.get(i)).getOrElse(obsNoInstrument);
+    }
+
+    public final ISPNodeInitializer<ISPObsExecLog, ObsExecLog> obsExecLog =
+            ObsExecLog.NI;
+
+    public final ISPNodeInitializer<ISPObsQaLog, ObsQaLog> obsQaLog =
+            ObsQaLog.NI;
+
+    public final ISPNodeInitializer<ISPProgram, SPProgram> program =
+            ProgramNI.instance;
+
+    public final ISPNodeInitializer<ISPNightlyRecord, NightlyRecord> record =
+            NightlyRecord.NI;
+
+    public final ISPNodeInitializer<ISPTemplateFolder, TemplateFolder> templateFolder =
+            TemplateFolder.NI;
+
+    public final ISPNodeInitializer<ISPTemplateGroup, TemplateGroup> templateGroup =
+            TemplateGroup.NI;
+
+    public final ISPNodeInitializer<ISPTemplateParameters, TemplateParameters> templateParameters =
+            TemplateParameters.NI;
+
+    public final Map<SPComponentType, ISPNodeInitializer<ISPObsComponent, ? extends ISPDataObject>> obsComp =
```

Comment:
Do we have a unit test asserting that these maps are sufficiently full? It would be an easy place to make an error in the future.